compiler/circle-mpqsolver/src/bisection/VISQErrorApproximator.cpp

```diff
@@ -17,23 +17,45 @@
 #include "VISQErrorApproximator.h"
 
 #include <fstream>
+#include <json.h>
 
 using namespace mpqsolver::bisection;
 
 void VISQErrorApproximator::init(const std::string &visq_data_path)
 {
   // read file
   std::ifstream file(visq_data_path);
-  if (!init(file))
+  try
+  {
+    init(file);
+  }
+  catch (std::runtime_error &)
   {
     throw std::runtime_error("Invalid visq file " + visq_data_path);
   }
 }
 
-bool VISQErrorApproximator::init(std::istream &)
+void VISQErrorApproximator::init(std::istream &visq_stream)
 {
-  // TODO
-  return true;
+  Json::Reader reader;
+  Json::Value completeJsonData;
+  if (!reader.parse(visq_stream, completeJsonData))
+  {
+    throw std::runtime_error("Invalid visq stream");
+  }
+
+  if (!completeJsonData.isMember("error"))
+  {
+    throw std::runtime_error("No error section in visq stream");
```

Comment:
optional) 
```suggestion
    throw std::runtime_error("No 'error' section in visq stream");
```

Reply:
Q) what does `section` mean?

Reply:
> Q) what does `section` mean?

`section` stands for serialized compound error info layerwise. but it can be 'array' or just `info`.